### PR TITLE
Adds missing methods for creating custom binary MediaTypes (#2008)

### PR DESCRIPTION
**Pull request by nikom1337**
and additional example for JAVA API (was missing in documentation).

Refs: #2008

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: docs/src/test/java/docs/http/javadsl/server/directives/JsonStreamingFullExample.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    public static void main(String[] args) throws Exception {
+        ActorSystem system = ActorSystem.create();
+        final JsonStreamingFullExample app = new JsonStreamingFullExample();
+        final Http http = Http.get(system);
+        final ActorMaterializer materializer = ActorMaterializer.create(system);
+
+        final Flow<HttpRequest, HttpResponse, NotUsed> routeFlow = app.createRoute().flow(system, materializer);
+        final CompletionStage<ServerBinding> binding = http.bindAndHandle(routeFlow, ConnectHttp.toHost("localhost", 8080), materializer);
```
Comment (raboof) [inline docs/src/test/java/docs/http/javadsl/server/directives/JsonStreamingFullExample.java]: I generally end examples at `http.bindAndHandle(routeFlow, ConnectHttp.toHost("localhost", 8080), materializer);` - for maven/gradle you can then `^C` to close the application, and on sbt you can use sbt-revolver.

File: docs/src/main/paradox/routing-dsl/source-streaming-support.md [hunk 2]
```diff
@@ -182,7 +180,8 @@ which we hope to add in a future release.
 Scala
 :   @@snip [JsonStreamingFullExamples.scala]($test$/scala/docs/http/scaladsl/server/directives/JsonStreamingFullExamples.scala) { #custom-content-type }
 
-@@@
+Java
+:   @@snip [JsonStreamingFullExample.java]($test$/java/docs/http/javadsl/server/directives/JsonStreamingFullExample.java) { #custom-content-type }
```
Comment (raboof) [inline docs/src/main/paradox/routing-dsl/source-streaming-support.md]: :+1:

File: docs/src/test/java/docs/http/javadsl/server/directives/JsonStreamingFullExample.java (hunk 1 above)
Comment (nikom1337) [reply docs/src/test/java/docs/http/javadsl/server/directives/JsonStreamingFullExample.java]: Ok, thanks. I'll fix that later today or tomorrow :)

**Comment by raboof** [comment]
Great, thanks!

**Pull request merged**